elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,205 @@
+package com.yahoo.elide.graphql;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.jsonapi.EntityProjectionMaker;
+import com.yahoo.elide.request.Argument;
+import com.yahoo.elide.request.Attribute;
+import com.yahoo.elide.request.EntityProjection;
+
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.CommonTokenStream;
+
+import graphql.GraphQL;
+import graphql.parser.antlr.GraphqlBaseVisitor;
+import graphql.parser.antlr.GraphqlLexer;
+import graphql.parser.antlr.GraphqlParser;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * <b>{@link GraphQLEntityProjectionMaker} produces {@link EntityProjection} that is only used in read operations</b>.
+ * Read-Write GraphQL query should be processed by GraphQL API instead({@link GraphQL#execute(String)}).
+ *
+ * @see EntityProjection
+ *
+ * TODO - transform sorting
+ * TODO - transform pagination
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class GraphQLEntityProjectionMaker extends GraphqlBaseVisitor<Void> {
```

Comment:
Although there are 2 approaches to map a GraphQL query to EntityProjection

1. query -> GraphQL Visitor -> Document -> EntityProjection, or
2. query -> GraphQL Visitor -> EntityProjection,

the 2nd approach is used, because

* The 1st approach has unnecessary overhead of constructing `graphql.language.Document`
* As an uncontrollable abstraction layer, `graphql.language.Document` is going to make it difficult to deeply integrate GraphQL into `EntityProjection` in the future
* If GraphQL has a breaking-change on language grammar/syntax, it is very likely that `GraphQLEntityProjectionMaker` will decompile, allowing us to make corresponding changes at compile time, rather than at runtime. This benefit is less likely by `graphql.language.Document`

Reply:
There is another approach too:

https://stackoverflow.com/questions/48909715/access-selection-set-of-a-graphql-query

From the Document (obtained from the parse tree), you can access the selection sets for each root query and just traverse those to build the EntityProjection.  That might be the simplest approach.

Reply:
There will be lots if-else type casting in this way and the if-else approach still binds closely to ANTLR grammar.